app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -500,17 +499,39 @@ public void onFailure(Call<EditResponse> call, Throwable t) {
 
     @SuppressWarnings("deprecation")
     @TargetApi(Build.VERSION_CODES.HONEYCOMB)
-    private void copyMessage(Message msg) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
-            android.content.ClipboardManager clipboard = (android.content.ClipboardManager) app
-                    .getSystemService(Context.CLIPBOARD_SERVICE);
-            clipboard.setPrimaryClip(ClipData.newPlainText("Zulip Message",
-                    msg.getContent()));
-        } else {
-            android.text.ClipboardManager clipboard = (android.text.ClipboardManager) app
-                    .getSystemService(Context.CLIPBOARD_SERVICE);
-            clipboard.setText(msg.getContent());
-        }
+    private void copyMessage(final Message message) {
+        final CommonProgressDialog commonProgressDialog = new CommonProgressDialog(getContext());
```

Comment:
(same comment) I think we might wanna override onFailure() to dismiss CommonProgressDialog as well since DefaultCallback doesn't take care of that (https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/networking/util/DefaultCallback.java#L30).